src/spec/EventSpec.ts

```diff
@@ -44,9 +44,22 @@ describe('An Event Dispatcher', () => {
     expect(eventFired).toBeTruthy();
   });
 
-  //it('has event handlers called in the right order', () => {
-  //   //TODO
-  //});
+  it('has event handlers called in the right order', () => {
+    const subscriptions = [];
+    const eventHistory = [];
+
+    for (let i = 0; i < 5; i++) {
+      subscriptions.push(i);
+    }
+
+    subscriptions
+      .sort(() => Math.random() - 0.5)
```

Comment:
Tests should always do exactly the same thing, so I think we should steer away from Math.random here.
To show that they're always called in the correct order, we could have the test change the order in a predictable way a couple of times and have `expect` assertions show that the order is correct after each one of those changes.

Reply:
I can definitely do that. I think testing with controlled random numbers is a viable option conceptually, if done correctly and in the right circumstance, but I'd like to conform to how you guys are currently doing things so I will gladly make the change.

Reply:
@alanag13 changes pushed to remove the random ordering.